docs/api_docs/maintenance/scale/horizontal-scale.md

```diff
@@ -0,0 +1,468 @@
+
+```yaml
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: Cluster
+metadata:
+  name: redis
+  namespace: kubeblocks-cloud-ns
+spec:
+  componentSpecs:
+  - name: proxy
+    componentDef: redis-proxy
+    replicas: 10
+    instances:
+    - name: proxy-2c
+      replicas: 3
+      resources:
+        limits:
+          cpu: 2
+          memory: 4Gi
+    resources:
+      limits:
+        cpu: 4
+        memory: 8Gi
+    offlineInstances:
+    - redis-proxy-proxy-2c4g-0
+    - redis-proxy-proxy-2c4g-1  
+```
+
+### Scaleout
+
+#### Example 1
+
+This example illustrates how to scale out replicas to 16 by applying an OpsRequest.
+
+The created instances are as follows:
+
+Three replicas equipped with 4C8G: `redis-proxy-7`, `redis-proxy-8`, `redis-proxy-9`
+Three replicas equipped with 2C4G: `redis-proxy-2c4g-5`, `redis-proxy-2c4g-6`, `redis-proxy-2c4g-7`
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: OpsRequest
+metadata:
+  generateName: ops-horizontalscaling
+spec:
+  type: HorizontalScaling
+  clusterName: redis
+  horizontalScaling:
+  - componentName: proxy
+    scaleOut: 
+      replicaChanges: 6
+      instances: 
+      - name: proxy-2c4g
+        replicasChanges: 3
+EOF
+```
+
+#### Example 2
+
+This example illustrates how to apply an OpsRequest to scale out replicas to 16 and three of the new replicas are proxies equipped with 8C16G.
+
+The created instances are as follows:
+
+4C8G: `redis-proxy-7`, `redis-proxy-8`, `redis-proxy-9`
+2C4G: `redis-proxy-8c16g-0`, `redis-proxy-8c16g-1`, `redis-proxy-8c16g-2`
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: OpsRequest
+metadata:
+  generateName: ops-horizontalscaling
+spec:
+  type: HorizontalScaling
+  clusterName: redis
+  horizontalScaling:
+  - componentName: proxy
+    scaleOut:
+      replicaChanges: 6 
+      newInstances:
+      - name: proxy-8c16g
+        replicas: 3
+        resources:
+          limits:
+            cpu: 8
+            memory: 16Gi  
+EOF
+```
+
+#### Example 3
+
+This example illustrates how to apply an OpsRequest to add the offline instances to the cluster. After the operation, there will be 12 replicas.
+
+The added pods in this example are `redis-proxy-2c4g-0` and `redis-proxy-2c4g-1`.
+
+```yaml
+kubectl apply -f - <<EOF
+apiVersion: apps.kubeblocks.io/v1alpha1
+kind: OpsRequest
+metadata:
+  generateName: ops-horizontalscaling
+spec:
+  type: HorizontalScaling
+  clusterName: redis
+  horizontalScaling:
+  - componentName: proxy
+    scaleOut:
+      offlineInstancesToOnline:
+      - redis-proxy-proxy-2c4g-0
+      - redis-proxy-proxy-2c4g-1     
+EOF   
+```
+
+### Scalein
+
+#### Why do you need to scale for specified instances
+
+KubeBlocks generated workloads as *StatefulSets*, which was a double-edged sword. While KubeBlocks could leverage the advantages of a *StatefulSets* to manage stateful applications like databases, it inherited its limitations.
+
```

Comment:
`StatefulSet` can only scale in starting from the pod ordinal in descending order. so we instead of it with `InstanceSet` and can scale in the specified pod name.